### PR TITLE
If ForceUserspaceProxy is true, check for iptables based proxy should not be performed.

### PR DESCRIPTION
**Pull request by aveshagarwal**
If ForceUserspaceProxy is true, check for iptables based proxy should not be performed.

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra

**Comment by caesarxuchao** [comment]
@kubernetes/goog-cluster

**Comment by aveshagarwal** [comment]
@k8s-bot test this

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -168,11 +168,16 @@ func (s *ProxyServer) Run(_ []string) error {
...
+	if !s.ForceUserspaceProxy {
+		var err error
+		// guaranteed false on error, error only necessary for debugging
+		shouldUseIptables, err = iptables.ShouldUseIptablesProxier()
+		if err != nil {
+			glog.Errorf("Can't determine whether to use iptables proxy, using userspace proxier: %v", err)
+		}
 	}
+
 	if !s.ForceUserspaceProxy && shouldUseIptables {
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go]: This can be reduced to `if shouldUseIptables {` can't it?

**Comment by thockin** [comment]
thanks for the cleanup, just one nit.

**Comment by thockin** [comment]
thanks for the cleanup, just one nit.

File: cmd/kube-proxy/app/server.go (hunk 1 above)
Comment (aveshagarwal) [inline cmd/kube-proxy/app/server.go]: Right, I will update the PR.

Comment (thockin) [inline cmd/kube-proxy/app/server.go]: ping me when pushed.

Comment (aveshagarwal) [inline cmd/kube-proxy/app/server.go]: Just pushed updated PR.

**Comment by thockin** [comment]
LGTM

**Comment by aveshagarwal** [comment]
Seems like it is waiting for ok-to-merge label.

**Pull request merged**